main.cpp

```diff
@@ -0,0 +1,165 @@
+#include <iostream>
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+#include "test/catch.hpp"
+#include "detect_current_range.h"
+#include "high_fidelity_current_sensor.h"
+#include "charging_discharging_current_sensor.h"
+
+vector<string> rangeOfReadings;
+uint32_t sizeOfcurrentInAmpere;
+uint32_t sizeOfcurrentSensor;
+vector<uint32_t> actual12BitCurrentValueInAmpere;
+vector<int32_t> actual10BitCurrentValueInAmpere;
+
+/*Start: Test Current Ranges*/
+
+TEST_CASE("test current set 1") {

```

Comment:
Good set of test cases :+1: 
There can be one more test case with a multiple ranges with atleast one range that has single value and atleast one range that has more than 1 values.